federation-js/src/composition/compose.ts

```diff
@@ -514,6 +523,7 @@ export function addFederationMetadataToSchemaNodes({
       // TODO: Why don't we need to check for non-object types here
       if (isObjectType(namedType)) {
         const field = namedType.getFields()[fieldName];
+        if (!field) break;
```

Comment:
`break` was a mistake here, but corrected in https://github.com/apollographql/federation/pull/481